cortex/utils.py

```diff
@@ -152,11 +165,18 @@ def generate_token(config, validity=2):
             "aud": config.get("audience"),
             "sub": config.get("username"),
         }
+
+        server_ts_dt = datetime.fromtimestamp(
+            _get_fabric_server_ts(config) / 1000, tz=tz.gettz("UTC")
+        )  # fabric info returns serverTs in milliseconds
+
+        expiry = server_ts_dt + timedelta(minutes=validity)
+
         token = py_jwt.generate_jwt(
```

Comment:
Missing issueAt / IAT ?

Reply:
Does this go into `other_headers` ?

Reply:
Compared to what we do in cli
```
    return new jose.SignJWT({})
        .setProtectedHeader({ alg: 'EdDSA', kid: jwk.kid })
        .setSubject(username)
        .setAudience(audience)
        .setIssuer(issuer)
        .setIssuedAt(Math.floor(serverTs / 1000)) // in seconds
        .setExpirationTime(expiry)
        .sign(jwtSigner, { kid: jwk.kid });
```

Think it ends up payload...  should be toplevel claim ...  https://pyjwt.readthedocs.io/en/stable/usage.html?highlight=generate#registered-claim-names. looking

Reply:
ok looks py_jwt sets this automatically

Reply:
claims['iat'] = timegm(now.utctimetuple()). Lame...  so if client's is  ahead/fast we'll have token issues

Reply:
yeah im setting it to serverts in the payload

Reply:
done